src/main/java/cz/kominekjan/disenchantment/guis/impl/EnchantmentsGUI.java

```diff
@@ -107,22 +104,23 @@ public final GUIItem[] getEnchantmentsItems(List<Enchantment> enchantments) {
 
             if (enchantment == null) continue;
 
-            EnchantmentState disenchantmentState = disenchantmentEnchantmentsStates.getOrDefault(enchantment, EnchantmentState.ENABLED);
-            EnchantmentState shattermentState = shattermentEnchantmentsStates.getOrDefault(enchantment, EnchantmentState.ENABLED);
+            EnchantmentState[] disenchantmentState = new EnchantmentState[]{disenchantmentEnchantmentsStates.getOrDefault(enchantment, EnchantmentState.ENABLED)};
+            EnchantmentState[] shattermentState = new EnchantmentState[]{shattermentEnchantmentsStates.getOrDefault(enchantment, EnchantmentState.ENABLED)};
```

Comment:
officially confused... why array?

Reply:
We need to change the stored EnchantmentState in a lambda as we only care about previous state and not the state at the gui creation.
lambda need final variable. that mean we can't just store a disenchantmentState and change it as the variable would not be considered final.
So, we either need: 
- An array to store the current object (the array will be final but not what stored inside). current implementation
- an AtomicReference<EnchantmentState>.  previously implemented but you deleted it so I guessed that was not a good idea.
- Any other type of final reference to a reference

Reply:
oh... well, in that case that atomic reference isn't bad :D

Reply:
Done